### PR TITLE
OSS testgrid dashboard group

### PR DESCRIPTION
**Pull request by krzyzacy**
Created `1.7-upgrade-all` to test with.

/assign @fejta @michelle192837

### CODE REVIEW

**Comment by fejta** [comment]
/unassign

Would like Michelle to review this. Please add me back if my attention is needed

File: testgrid/config/config_test.go [hunk 1]
```diff
@@ -145,6 +145,14 @@ func TestConfig(t *testing.T) {
 		}
 	}
 
+	for _, dashboardGroup := range config.DashboardGroups {
+		for _, dashboard := range dashboardGroup.DashboardNames {
+			if ok, _ := dashboardmap[dashboard]; !ok {
+				t.Errorf("Dashboard %v need to be defined before add to a dashboard group!", dashboard)
```
Comment (michelle192837) [inline testgrid/config/config_test.go]: nit: *needs *adding

Comment (krzyzacy) [inline testgrid/config/config_test.go]: done

File: testgrid/config/config_test.go [hunk 2]
```diff
@@ -145,6 +145,14 @@ func TestConfig(t *testing.T) {
 		}
 	}
 
+	for _, dashboardGroup := range config.DashboardGroups {
```
Comment (michelle192837) [inline testgrid/config/config_test.go]: Can you also add tests to 1) ensure no dashboard names are duplicated in a group, and 2) there aren't duplicated dashboard group names?

Comment (fejta) [inline testgrid/config/config_test.go]: Good test

Comment (krzyzacy) [inline testgrid/config/config_test.go]: good call, added

**Comment by michelle192837** [comment]
/lgtm

Thanks for adding this!

**Comment by krzyzacy** [comment]
let's see if it will work out

**Pull request merged**